joomla/joomla-cms
# [5.3] Redesign carousel implementation to reflect documentation - bug fix

### Summary of Changes
This pull request fixes a bug in the Bootstrap Caraousel implementation of Joomla! (PHP and ES/JS) that causes valid parameters to be removed from the JavaScript options object that is fed to the Carousel instance.

### Testing Instructions
I do not know if the Patch Tester extension will do the JavaScript re-build / pre-processing for you, and hope you know best what to do in this regard.

- Activate the debug mode in the global configuration to better be able to inspect the generated JavaScript option object in the source code (Browser: Ctrl + U).
- Install, activate and publish the demo module provided with the pull request on a module position of your choice (e.g. Cassiopeia: main-top). Make sure to set the Menu / Module Assignent to "On all pages".
- Open / refresh the page and change some params in the module manager (e.g. keyboard | wrap to false).

### Actual result BEFORE applying this Pull Request
All options set to **_false_** are filtered out, causing the carousel to fall back to its default values.

### Expected result AFTER applying this Pull Request
Params that can have a **_false_** value are recognised and work as expected.

### Link to documentations
Please select:
- [X] Documentation link for docs.joomla.org: https://docs.joomla.org/J4.x:Using_Bootstrap_Components_in_Joomla_4#Carousel (slide vs ride)


### Demo Module
[mod_bs_carousel.zip](https://github.com/joomla/joomla-cms/files/9744346/mod_bs_carousel.zip)
